### DISCUSSION TITLE
Please, help me with DateTime LineSeries mapping

### DISCUSSION THREAD

**Discussion by toxicabit** [Live-Charts/LiveCharts2#1532] [Q&A]
Can someone explain me how do I properly map datetime property of my custom class to X axis.
I have a custom data type like 
```
public class MyClass 
{
     public int Id { get; set; }
     public double Data { get; set; }
     public DateTime Date { get; set; }
}
```
And I want to create a CartesianChart with Date on X and Data on Y.
So I do
```
Series = new ISeries[]
{
    new LineSeries<MyClass>
    {
        Values = instance,
        Mapping = (data, _) => new(data.Date, data.Data)
    }
};
```
But Coordinate requires (double, double) constructor.
Can you tell me how to map DateTime, please

**Comment by WanftMoon** [comment]
You can just use a DateTimePoint.
```
Values = new ObservableCollection<DateTimePoint>(
        Datapoints.Select(dp => new DateTimePoint(dp.DateTime, dp.Value))
)
```

**Comment by Benny-prog** [comment]
Hi, I have a quite similar requirement. As I understand, if I use the suggestion from @WanftMoon, I would also need to use 'DateTimeAxis' as XAxis, correct?
My problem is, I want to be able to hide empty time slots from my chart, which I didn't manage with the usage of 'DateTimeAxis'.

**Reply by beto-rodriguez** [reply]
@Benny-prog then maybe you don't need to use the DateTime type, instead just plot double values, and use the Axis.Labels property to show the dates.

**Reply by Benny-prog** [reply]
@beto-rodriguez, thanks for your quick reply.
I'm having 4 series, which can also have no value at a point of time. 
e.g. I want to display 30 days and series A doesn't have a value at day 2. Then all values from day 3 on are displayed one day before (day 3 at day 2, day 4 at day 3 and so on).
Would you suggest to add zero value at day 2 or is there any other possibility?

**Reply by WanftMoon** [reply]
> @beto-rodriguez, thanks for your quick reply. I'm having 4 series, which can also have no value at a point of time. e.g. I want to display 30 days and series A doesn't have a value at day 2. Then all values from day 3 on are displayed one day before (day 3 at day 2, day 4 at day 3 and so on). Would you suggest to add zero value at day 2 or is there any other possibility?

not sure exactly how you are modeling or which kind of chart you are using. but it sounds like either you want to play with the visibility of the series, or you want to introduce gaps in it.

have you looked at this?
https://livecharts.dev/docs/Maui/2.0.0-rc2/samples.general.nullPoints

**Reply by Benny-prog** [reply]
Thank you very much, with a combination of your comments, I was able to get the requested behaviour.
